### PR TITLE
Postgres notifier

### PR DESCRIPTION
**Pull request by donatello**
Ready for review.

### CODE REVIEW

File: cmd/config-migrate.go [hunk 1]
```diff
@@ -434,6 +435,11 @@ func migrateV7ToV8() error {
 	} else {
 		srvConfig.Notify.Redis = cv7.Notify.Redis
 	}
+	if len(cv7.Notify.PostgreSQL) == 0 {
```
Comment (harshavardhana) [inline cmd/config-migrate.go]: This is dangerous and should be avoided,  implement a proper config migration. do not edit configs like this.  Migrate to v9.

File: cmd/notify-postgresql.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+			"Records": entry.Data["Records"],
+		})
+		if err != nil {
+			return fmt.Errorf(
+				"Unable to encode event %v to JSON - got error - %v",
+				entry.Data["Records"], err,
+			)
+		}
+
+		// attempt to insert into the table
```
Comment (harshavardhana) [inline cmd/notify-postgresql.go]: Why does this say "attempt" ? - isn't this atomic?. Does this retry transparently if the attempt fails?

File: cmd/notify-postgresql.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+	// TODO: check that table exists?
+
+	// create prepared statements
+	stmts := make(map[string]*sql.Stmt)
+	// insert or update statement
+	stmts["upsertRow"], err = db.Prepare(fmt.Sprintf(upsertRow, pgN.Table))
+	if err != nil {
+		return pgConn{}, errors.New("Unable to create upsert prepared statement for postgres notification.")
+	}
+	stmts["deleteRow"], err = db.Prepare(fmt.Sprintf(deleteRow, pgN.Table))
```
Comment (harshavardhana) [inline cmd/notify-postgresql.go:167]: Use the same error or embed this into a newly formed error. using `fmt.Errorf`

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/config-migrate.go (hunk 1 above)
Comment (donatello) [reply cmd/config-migrate.go]: Fixed, but updating the test code.

File: cmd/notify-postgresql.go (hunk 3 above)
Comment (donatello) [reply cmd/notify-postgresql.go:167]: Fixed, and tried to make errors in this section uniform with `fmt.Errorf`

File: cmd/notify-postgresql.go (hunk 2 above)
Comment (donatello) [reply cmd/notify-postgresql.go]: Fixed this comment. It is atomic, and failures are only related to connection problems and other un-controllables.

**Comment by donatello** [comment]
Going to add table creation as well if table is not present at start up.

**Comment by donatello** [comment]
@harshavardhana Now table is created if it does not exist at startup.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**